hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/util/PartitionedGSet.java

```diff
@@ -280,12 +279,22 @@ private void printStats() {
 
     public EntryIterator() {
       keyIterator = partitions.keySet().iterator();
-      K curKey = partitions.firstKey();
-      partitionIterator = getPartition(curKey).iterator();
+ 
+      if (!keyIterator.hasNext()) {
+        partitionIterator = null;
+        return;
+      }
+
+      K firstKey = keyIterator.next();
+      partitionIterator = partitions.get(firstKey).iterator();
     }
 
     @Override
     public boolean hasNext() {
+
+      if (partitionIterator == null)
```

Comment:
So if somebody adds a partition after `EntryIterator()` for the empty `PartitionedGSet` was created, it will never be picked up. Should we try to set `partitionIterator` if it is null.

Reply:
I don't think we need to handle that case: if one creates the Iterator() for an empty PartitionedGSet, it makes sense the iterator returns zero elements which represents a correct snapshot.